trade-finance-manager-api/src/v1/middleware/validate-sso-feature-flag.test.ts

```diff
@@ -0,0 +1,86 @@
+import { isTfmSsoFeatureFlagEnabled } from '@ukef/dtfs2-common';
+import { HttpStatusCode } from 'axios';
+import { RequestHandler } from 'express';
+import httpMocks from 'node-mocks-http';
+import { validateSsoFeatureFlagIsOff, validateSsoFeatureFlagIsOn } from './validate-sso-feature-flag';
+
+jest.mock('@ukef/dtfs2-common', () => ({
+  isTfmSsoFeatureFlagEnabled: jest.fn(),
+}));
+
+describe('validateSsoFeatureFlag', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('validateSsoFeatureFlagIsOn', () => {
+    const makeRequest = validateSsoFeatureFlagIsOn;
+
+    describe('when the SSO feature flag is disabled', () => {
+      beforeEach(() => {
+        mockIsTfmSsoFeatureFlag(false);
+      });
+
+      itReturnsAnErrorResponse({ makeRequest });
+    });
+
+    describe('when the SSO feature flag is enabled', () => {
+      beforeEach(() => {
+        mockIsTfmSsoFeatureFlag(true);
+      });
+
+      itCallsTheNextMiddleware({
+        makeRequest,
+      });
+    });
+  });
+
+  describe('validateSsoFeatureFlagIsOff', () => {
+    const makeRequest = validateSsoFeatureFlagIsOff;
+
+    describe('when the SSO feature flag is disabled', () => {
+      beforeEach(() => {
+        mockIsTfmSsoFeatureFlag(false);
+      });
+
+      itCallsTheNextMiddleware({
+        makeRequest,
+      });
+    });
+
+    describe('when the SSO feature flag is enabled', () => {
+      beforeEach(() => {
+        mockIsTfmSsoFeatureFlag(true);
+      });
+
+      itReturnsAnErrorResponse({ makeRequest });
+    });
+  });
+
+  function getHttpMocks() {
+    return httpMocks.createMocks();
+  }
+
+  function mockIsTfmSsoFeatureFlag(value: boolean) {
+    jest.mocked(isTfmSsoFeatureFlagEnabled).mockReturnValue(value);
+  }
+
+  function itCallsTheNextMiddleware({ makeRequest }: { makeRequest: RequestHandler }) {
+    it('calls the next middleware', () => {
+      const { req, res } = getHttpMocks();
+      const next = jest.fn();
+      makeRequest(req, res, next);
+      expect(next).toHaveBeenCalled();
+    });
+  }
+
+  function itReturnsAnErrorResponse({ makeRequest }: { makeRequest: RequestHandler }) {
```

Comment:
I like the DRYness of these functions but I think it overcomplicates the test, found it a bit tricky to follow, I think something like this would be nicer

```ts
describe('validateSsoFeatureFlag', () => {
  const { req, res } = httpMocks.createMocks();

  const next = jest.fn();

  beforeEach(() => {
    jest.resetAllMocks();
  });

  describe('when the SSO feature flag is enabled', () => {
    beforeEach(() => {
      mockIsTfmSsoFeatureFlag(true);
    });

    it('should call the next middleware', () => {
      expect(next).toHaveBeenCalled();
    });
  });
});
```